mono/mini/interp/transform.c

```diff
@@ -7245,15 +7245,28 @@ interp_cprop (TransformData *td)
 				InterpInst *obj_ins = sp [-2].ins;
 				if (obj_ins->opcode == MINT_LDLOC_O) {
 					int loc_index = obj_ins->data [0];
-					int fld_offset = ins->data [0];
-					int mt = ins->opcode - MINT_STFLD_I1;
-					ins = interp_insert_ins (td, ins, MINT_STLOCFLD_I1 + mt);
-					ins->data [0] = loc_index;
-					ins->data [1] = fld_offset;
-					interp_clear_ins (td, ins->prev);
-					interp_clear_ins (td, obj_ins);
-					mono_interp_stats.super_instructions++;
-					mono_interp_stats.killed_instructions++;
+					gboolean valid = TRUE;
+
+					/* this optimization delays the load of the local, which we only can do if it hasn't been written to since then */
+					for (InterpInst *cur = obj_ins; valid && cur && cur != ins; cur = cur->next) {
+						guint16 op = cur->opcode;
```

Comment:
Can't this take a long time ?

Reply:
on only operates on basic blocks afaik

Reply:
This is not the way to solve it, it is dirty. When we get to emitting this superinstruction we check if obj_ins != NULL which means that we know what value is in that stack slot. From what I understood from you, the value on the stack is no longer up to date so it should have been cleared (see `clear_stack_content_info_for_local`). We could also replace it with the updated local (I don't remember if we do this currently). You should first check why the instruction that modifies the local value doesn't also update the stack slots where this local was pushed